### PR TITLE
STORM-2327 ConfigurableTopology

### PR DESCRIPTION
**Pull request by jnioche**
See https://issues.apache.org/jira/browse/STORM-2327

### CODE REVIEW

**Comment by HeartSaVioR** [comment]
@jnioche 
Thanks for the contribution. It looks good.
Btw, it would be great if we apply ConfigurableTopology to all of topology implementations on example directory or at least storm-starter to expose this class to more users. What do you think?

**Comment by jnioche** [comment]
Thanks @HeartSaVioR. Good idea, I'll have a look at the topology classes in storm-starter. Just came across the (StormRunner)[https://github.com/apache/storm/blob/master/examples/storm-starter/src/jvm/org/apache/storm/starter/util/StormRunner.java] class, which the ConfigurableTopology could replace.

**Comment by HeartSaVioR** [comment]
@jnioche 
Yes I also think you can replace StormRunner with ConfigurableTopology. 
Btw, there're lots of topologies so you might want to only apply some of them. If that's the case please let me know. I think that would be also OK.

**Comment by jnioche** [comment]
@HeartSaVioR I removed StormRunner and modified some of the topologies so that they extend ConfigurableTopology. Please let me know your thoughts on this.

File: examples/storm-starter/src/jvm/org/apache/storm/starter/RollingTopWords.java [hunk 1]
```diff
@@ -17,114 +17,89 @@
...
+   * "slidingWindowCounts".
    *
    * Examples:
    *
    * ```
    *
-   * # Runs in local mode (LocalCluster), with topology name "slidingWindowCounts"
-   * $ storm jar storm-starter-jar-with-dependencies.jar org.apache.storm.starter.RollingTopWords
+   * # Runs in local mode (LocalCluster), with topology name
+   * "slidingWindowCounts" $ storm jar storm-starter-jar-with-dependencies.jar
```
Comment (HeartSaVioR) [inline examples/storm-starter/src/jvm/org/apache/storm/starter/RollingTopWords.java]: needs new line before `$`, and next line should be added after `jar`.

Comment (HeartSaVioR) [inline examples/storm-starter/src/jvm/org/apache/storm/starter/RollingTopWords.java]: same things are repeated below

**Comment by HeartSaVioR** [comment]
@jnioche 
Could you also modify README.md on storm-starter to apply the change for topologies? Looks great otherwise.

File: examples/storm-starter/src/jvm/org/apache/storm/starter/RollingTopWords.java (hunk 1 above)
Comment (jnioche) [inline examples/storm-starter/src/jvm/org/apache/storm/starter/RollingTopWords.java]: done!

**Comment by HeartSaVioR** [comment]
+1 
@jnioche 
Could you squash commits into one? `STORM-2327 ConfigurableTopology` would be good as commit title for squashed commit.

**Comment by jnioche** [comment]
I made a mess with this branch and have pushed a clean, squashed PR as a replacement -> #1907

**Pull request closed without merging**